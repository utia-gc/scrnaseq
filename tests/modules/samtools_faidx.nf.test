nextflow_process {

    name "Test Process samtools_faidx"
    script "modules/samtools_faidx.nf"
    process "samtools_faidx"

    test("samtools_faidx generates fasta index for genome fasta") {
        // instantiate helper objects
        def refs = new RefsEnsembl()

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            process {
                """
                input[0] = file("${projectDir}/${refs.getGenome()}")
                """
            }
        }

        then {
            // test process.success
            assert process.success

            // test fasta index produced
            with(process.out.fai) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa.fai/
            }

            // test reproducibility
            assert snapshot(process.out).match()
        }

    }

}
