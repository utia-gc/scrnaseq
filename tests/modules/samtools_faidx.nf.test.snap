{
    "samtools_faidx generates fasta index for genome fasta": {
        "content": [
            {
                "0": [
                    "genome_I.fa.fai:md5,2cd76d936cbfa386b14154506c2041b2"
                ],
                "fai": [
                    "genome_I.fa.fai:md5,2cd76d936cbfa386b14154506c2041b2"
                ]
            }
        ],
        "timestamp": "2023-09-27T15:36:50+0000"
    }
}